merge_and_rename_NGI_fastq_files.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env/python
+# Merges all fastq_files from each samples into one file. 
+# ARGV 1 Base directory with files to be merged
+# ARGV 2 Destination dir of the merged output files
+
+import re
+import os
+import sys
+import shutil
+import argparse
+import collections
+def merge_files(input_dir, dest_dir):
+   #Gather all fastq files in inputdir and its subdirs
+   fastq_files=[]
+   for subdir, dirs, files in os.walk(input_dir):
+       for fastq in  files:
+           if fastq.endswith('.fastq.gz'):
+               fastq_files.append(os.path.join(subdir, fastq))
+   
+    
+    #Match NGI sample number from flowcell
+    sample_pattern=re.compile("^(P[0-9]+_[0-9]+)_S[0-9]+_.+_R([1-2])_")
+    #Remove files that already have the right name (i.e have been merged already)
+    matches=[]
+    for fastq_file in fastq_files:
+        try:
+            match=sample_pattern.search(fastq_file).group(1)
+            if match:
+                matches.append(fastq_file)
+        except AttributeError: 
+            continue
+    fastq_files=matches
+   
+    ## Sort on basename to avoid collison
+    
+    while fastq_files:
+        tomerge=[]
+        
+        #grab one sample to work on
+        first=fastq_files[0]
+        fq_bn=os.path.basename(first)
+        sample_name=sample_pattern.match(fq_bn).group(1)
+        fastq_files_read1=[]
+        fastq_files_read2=[]
+        def getKey(item):
+                return item[1]
+        #make a list of lists with abs path and basename
+        for fq in fastq_files:
+            if sample_name in os.path.basename(fq) and "_R1_" in os.path.basename(fq):
+                fastq_files_read1.append([fq, os.path.basename(fq)])
+                
+            if sample_name in os.path.basename(fq) and "_R2_" in os.path.basename(fq):
+                fastq_files_read2.append([fq, os.path.basename(fq)])
+        #sort both lists on the basename.  
+        fastq_files_read1.sort(key=getKey)
+        fastq_files_read2.sort(key=getKey)
+        #Split the lists
+        fastq_files_read1, throw_away = zip(*fastq_files_read1)
```

Comment:
we can discuss about this tomorrow, but I think you can completely avoid to create a tuple and simply used `fq` on lines 50 and 53. Then line 55 and 56 become a simple sort() and you can completely skip this one.